orttraining/orttraining/core/graph/zero_optimizer_graph_builder.cc

```diff
@@ -167,6 +279,36 @@ static Status AddViewForParameters(
   return Status::OK();
 }
 
+static Status ModifyParametersForOptimizerPartitioningByBoundary(
+    const OptimizerGraphConfig& opt_graph_config,
+    std::vector<ArgDef>& gradient_argdefs,
+    std::vector<int64_t>& partitions,
+    int64_t& max_group_size) {
+  std::vector<int64_t> size_arr;
+  int max_size = 0;
+  int64_t total_size = 0;
+  for (size_t i = 0; i < gradient_argdefs.size(); i++) {
+    ArgDef gradient_argdef = gradient_argdefs[i];
+    ORT_ENFORCE(gradient_argdef.type_proto != nullptr);
+    const auto& gradient_shape_proto = gradient_argdef.type_proto->tensor_type().shape();
+    const TensorShape& gradient_shape = utils::GetTensorShapeFromTensorShapeProto(gradient_shape_proto);
+
+    total_size += int(gradient_shape.Size());
```

Comment:
a cast isn't needed here right?
also for casts, see https://google.github.io/styleguide/cppguide.html#Casting